File: ruby/acronym/acronym.rb
class Acronym
  WORD_BOUNDARIES = /\w+/

  def self.abbreviate(phrase)
    @phrase = phrase
    words.map do |word|
      first_letter(word)
    end.join
  end

  private

  def self.words
    @phrase.scan(WORD_BOUNDARIES)
  end

  def self.first_letter(word)
    word.upcase.split('').first
  end

end

File: csharp/acronym/Acronym.cs
using System;
using System.Text.RegularExpressions;

public static class Acronym
{

    public static string Abbreviate(string phrase)
    {
        string pattern = "([A-Z])[A-Z']*";
        string result = String.Join("", Regex.Split(phrase.ToUpper(), pattern));
        return Regex.Replace(result, @"[^A-Z]", "");
    }
}